baselines/pFedHN/pFedHN/strategy.py

```diff
@@ -0,0 +1,211 @@
+"""Strategy of the Federated Learning."""
+
+from collections import OrderedDict
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+import flwr as fl
+import torch
+from flwr.common import (
+    EvaluateIns,
+    EvaluateRes,
+    FitIns,
+    FitRes,
+    NDArrays,
+    Parameters,
+    Scalar,
+    ndarrays_to_parameters,
+    parameters_to_ndarrays,
+)
+from flwr.server.client_manager import ClientManager
+from flwr.server.client_proxy import ClientProxy
+
+from pFedHN.models import CNNHyper
+
+
+# pylint: disable=invalid-name
+# pylint: disable=too-many-instance-attributes
+class pFedHN(fl.server.strategy.Strategy):
+    """Federated strategy with pFedHN."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        config,
+        fraction_fit: float = 0.0,
+        fraction_evaluate: float = 1.0,
+        min_fit_clients: int = 1,
+        min_evaluate_clients: int = 1,
+        min_available_clients: int = 1,
+        accept_failures: bool = True,
+        initial_parameters: Optional[Parameters] = None,
+        evaluate_fn: Optional[
+            Callable[
+                [int, NDArrays, Dict[str, Scalar]],
+                Optional[Tuple[float, Dict[str, Scalar]]],
+            ]
+        ] = None,
+    ) -> None:
+        super().__init__()
+        self.fraction_fit = fraction_fit
+        self.fraction_evaluate = fraction_evaluate
+        self.min_fit_clients = min_fit_clients
+        self.min_evaluate_clients = min_evaluate_clients
+        self.min_available_clients = min_available_clients
+        self.accept_failures = accept_failures
+        self.initial_parameters = initial_parameters
+        self.cfg = config
+        self.evaluate_fn = evaluate_fn
+        # Initialising our hnet model in server
+        self.hnet = CNNHyper(
+            n_nodes=self.cfg.client.num_nodes,
+            embedding_dim=int(1 + self.cfg.client.num_nodes / 4),
```

Comment:
maybe worth having this computed above the `self.hnet` definition and point to where in the paper is it done?
just an idea.

Reply:
potentially, the `/4` could come from the config (in case people running your code want to change this -- even though in the paper is fixed to `/4`)

Reply:
Gave explanation both in base.yaml as well as server.py